projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/execution/JobExecutionResultManager.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.execution;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import com.google.gson.Gson;
+import lombok.Builder;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import com.vmware.taurus.service.KubernetesService;
```

Comment:
I thought we agreed, we'd move JobExecution out of KubernetesService?

Reply:
I don't think we should do it as part of this change. In general, we need to use DTO classes that describe Kubernetes objects like the rest of DTOs in KubernetesService. We also need to move much of the `getJobExecutionStatus` logic out of the class.